qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/parser/template/sections/FormSection.java

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+* Copyright (c) 2023 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.parser.template.sections;
+
+import java.util.List;
+
+import com.redhat.qute.parser.template.ASTVisitor;
+import com.redhat.qute.parser.template.Section;
+import com.redhat.qute.parser.template.SectionKind;
+import com.redhat.qute.services.diagnostics.CollectHtmlInputNamesVisitor;
+
+/**
+ * Form section.
+ * 
+ */
+public class FormSection extends CustomSection {
```

Comment:
After reading your code, I think it is not a good idea to have a FormSection which override the section kind and which breaks several tests because code is using CUSTOM.

More you store in the cache the input names, but we don't need to cache it, since it is used only for validation. 

Please remove this FormSection class.